### PR TITLE
Lint tweaks

### PR DESCRIPTION
**Pull request by XhmikosR**
This fixes #62 properly.

Check the non-whitespace diff https://github.com/feimosi/baguetteBox.js/compare/master...XhmikosR:lint?w=1

Personally, I'd get rid of the inline suppressions too.

Next step is to add ESLint, but I'm not very familiar with Gulp.

### CODE REVIEW

File: src/baguetteBox.js [hunk 1]
```diff
@@ -395,45 +422,50 @@
...
+        if (options.async && callback) {
             callback();
+        }
     }
 
     // Get image source location, mostly used for responsive images
     function getImageSrc(image) {
         // Set default image path from href
-        var result = imageElement.href;
+        var result = image.href;
```
Comment (XhmikosR) [inline src/baguetteBox.js]: @feimosi: this one I use `image` instead of `imageElement` since that's what I thought you meant to use.

**Comment by feimosi** [comment]
Finally, the project got some linting, thanks! I'll switch to ESLint in the future, JSHint is fine for now.

**Comment by XhmikosR** [comment]
I would try enabling the disabled checks first like latedef etc. We also need to find a proper way for the AMD check because those are reported as undefined.

**Comment by XhmikosR** [comment]
Also, how about disabling the inline suppressions? It's more code but readability is better IMO.

**Comment by feimosi** [comment]
Ok, I've just analysed the code, so:
- you can place a new option in the config : `"predef": ["define", "module"]` for globals
- we can't do much about `latedef` because it also considers ordinary functions, which is irrational (the definition is hoisted too) and unreadable (I prefer to put most outer functions at the top)
- all the inline code are polyfills, not written by me. I'd rather leave them minified and inlined or put the original readable and unminified versions.
- you can enable `"strict"` and put one where it's needed

**Comment by XhmikosR** [comment]
How about I'll just add a global for the undefined var instead of the predef. I'll leave the rest to you.

**Comment by XhmikosR** [comment]
As for the inline suppressions, that is why I suggested that you move the polyfills to separate files and just concat them on build; should be clearer.

PS. We really need `gulp lint` or something :)

**Comment by feimosi** [comment]
What do you mean by that: _just add a global for the undefined var_ ?

you're right, I'll think about extracting polyfills later. 
`gulp lint` will do that too :)

**Comment by XhmikosR** [comment]
I added the global directives in the last patch. The predef solution will
make those everywhere available so I prefer to be more in control.

**Comment by feimosi** [comment]
Good point :)

**Pull request merged**